.github/workflows/ci.yml

```diff
@@ -137,3 +137,68 @@ jobs:
       - uses: actions/checkout@v4
       - name: Build Docker Image
         run: docker build --target web -t ons .
+
+  functional-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        browser:
+          - chromium
+    needs:
+      - lint
+      - lint-front-end
+      - compile_static
+
+    env:
+      DJANGO_SETTINGS_MODULE: cms.settings.functional_test
+      SECRET_KEY: fake_secret_key_to_run_tests # pragma: allowlist secret
+      PLAYWRIGHT_BROWSER: ${{ matrix.browser }}
+      PLAYWRIGHT_TRACES_DIR: playwright_traces
+      PLAYWRIGHT_TRACE: 'true'
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Poetry
+        run: pipx install poetry==${{ env.POETRY_VERSION }}
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version-file: .python-version
+          cache: poetry
+
+      - name: Install dependencies
+        run: make install-dev
+
+      - name: Playwright Install
+        run: poetry run python -m playwright install --with-deps ${{ matrix.browser }}
+
+      - uses: actions/download-artifact@v4
+        with:
+          name: static
+          path: cms/static_compiled/
+
+      - name: Docker Compose Up
+        run: make functional-tests-up
+
+      - name: Fetch Design System templates
+        run: make load-design-system-templates
+
+      - name: Collect Static Files
+        run: poetry run ./manage.py collectstatic --verbosity 0 --noinput --clear
+
+      - name: Prepare Database
+        run: poetry run ./manage.py migrate --noinput
```

Comment:
Question: Is the need to run this command related to the bug I found? If so I'd recommend linking it so it's obvious it can be removed.

Since it looks like the test runner uses the test database instance, there _shouldn't_  be a need to run the real migrations

Reply:
Possibly, I believed the way it worked was the test runner copies the configured default database structure so pre-running migrations was required, but it might be related to the bug.

Reply:
It looks like it is related, so I'm adding those comments